pkg/eventing/manager_test.go

```diff
@@ -0,0 +1,531 @@
+				),
+			},
+			expectedResult: &appsv1.Deployment{},
+			expectedError:  nil,
+		},
+		{
+			name: "CreateOrUpdatePublisherProxy error",
+			eventing: testutils.NewEventingCR(
+				testutils.WithEventingCRName("test-eventing"),
+				testutils.WithEventingCRNamespace(ecdeployment.PublisherNamespace),
+				testutils.WithEventingInvalidBackend(),
+				testutils.WithEventingPublisherData(2, 4, "100m", "256Mi", "200m", "512Mi"),
+			),
+			givenNats: []natsv1alpha1.NATS{
+				*natstestutils.NewNATSCR(
+					natstestutils.WithNATSCRName("test-eventing"),
+					natstestutils.WithNATSCRNamespace(ecdeployment.PublisherNamespace),
+					natstestutils.WithNATSStateReady(),
+				),
+			},
+			expectedResult: nil,
+			expectedError:  fmt.Errorf("NATs backend is not specified in the eventing CR"),
+		},
+	}
+
+	// Iterate over the test cases and run sub-tests
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// given
+			ctx := context.Background()
+			mockECReconcileClient := new(mocks.ECReconcilerClient)
+			mockClient := new(mocks.Client)
+			kubeClient := new(k8smocks.Client)
+
+			kubeClient.On("GetNATSResources", ctx, tc.eventing.Namespace).Return(&natsv1alpha1.NATSList{
+				Items: tc.givenNats,
+			}, nil)
+			mockECReconcileClient.On("CreateOrUpdatePublisherProxy",
+				mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(&appsv1.Deployment{}, nil)
+
+			logger, _ := logger.New("json", "info")
+			em := EventingManager{
+				Client:             mockClient,
+				kubeClient:         kubeClient,
+				ecReconcilerClient: mockECReconcileClient,
+				logger:             logger,
+			}
+
+			// when
+			result, err := em.CreateOrUpdatePublisherProxy(ctx, tc.eventing)
+
+			// then
+			require.Equal(t, tc.expectedResult, result)
+			require.Equal(t, tc.expectedError, err)
+		})
+	}
+}
+
+func Test_CreateOrUpdateHPA(t *testing.T) {
+	// Define a list of test cases
+	testCases := []struct {
+		name              string
+		deployment        *appsv1.Deployment
+		eventing          *v1alpha1.Eventing
+		cpuUtilization    int32
+		memoryUtilization int32
+		givenGetErr       error
+		expectedError     error
+	}{
+		{
+			name: "Create new HPA",
+			deployment: &appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-deployment",
+					Namespace: "test-namespace",
+				},
+				Spec: appsv1.DeploymentSpec{
+					Replicas: int32Ptr(2),
+				},
+			},
+			eventing: testutils.NewEventingCR(
+				testutils.WithEventingCRName("test-eventing"),
+				testutils.WithEventingCRNamespace("test-namespace"),
+				testutils.WithEventingInvalidBackend(),
+				testutils.WithEventingPublisherData(1, 5, "100m", "256Mi", "200m", "512Mi"),
+			),
+			cpuUtilization:    50,
+			memoryUtilization: 50,
+			givenGetErr:       apierrors.NewNotFound(autoscalingv2.Resource("HorizontalPodAutoscaler"), "eventing-publisher-proxy"),
+			expectedError:     nil,
+		},
+		{
+			name: "Update existing HPA",
```

Comment:
I assume this test case is dependant on the previous test case, right?
If so, I believe test cases should not depend on each other.

Reply:
no it is not. If the not found error exists it creates otherwise it uppdates. As you [here](https://github.com/muralov/eventing-manager/blob/bb7b8867cac3987bac279f5103739f19988f4b8e/pkg/eventing/manager_test.go#L140C10-L140C10), create case have not found error.